middle_end/closure/closure_middle_end.ml

```diff
@@ -35,14 +35,23 @@ let lambda_to_clambda ~backend ~filename:_ ~prefixname:_ ~ppf_dump
     Closure.intro ~backend ~size:lambda.main_module_block_size lambda.code
   in
   let provenance : Clambda.usymbol_provenance =
+    let current_unit_ident =
+      Compilation_unit.get_current_exn ()
+      |> Compilation_unit.name
+      |> Compilation_unit.Name.to_string
+      |> Ident.create_persistent
+    in
     { original_idents = [];
-      module_path =
-        Path.Pident (Ident.create_persistent (Compilenv.current_unit_name ()));
+      module_path = Path.Pident current_unit_ident;
     }
   in
+  let symbol =
+    Symbol.for_current_unit ()
```

Comment:
Maybe this function should be called `Symbol.module_block_for_current_unit` or something?  Not sure.

Reply:
If we resurrect `Linkage_name`, then `Linkage_name.for_current_unit ()` would be pretty reasonable.

Reply:
I would like to define `Linkage_name.for_current_unit`, but currently that would entail a dependency loop with `Symbol` and `Compilation_unit`. (We could move some code from `Symbol` to `Linkage_name` to address this; I left a CR-someday.)

Reply:
Ah, I was trying to emphasise the module block aspect.

Reply:
Let's just leave this